File: back-end/GeekSpot.Domain/Entities/ItemTipo.cs
using System.ComponentModel.DataAnnotations;
using static GeekSpot.Utils.Biblioteca;

namespace GeekSpot.Domain.Entities
{
    public class ItemTipo
    {
        [Key]
        public int ItemTipoId { get; set; }
        public string? Tipo { get; set; } = null;
        public string? Descricao { get; set; } = null;
        public int IsNovoTipo { get; set; } = 0;
        public int IsAtivo { get; set; } = 1;
        public DateTime DataRegistro { get; set; } = HorarioBrasilia();
    }
}

File: front-end/utils/types/itemTipo.ts
export default interface iItemTipo {
    itemTipoId: number;
    tipo: string;
    descricao: string | null;
    isNovoTipo: number;
    isAtivo: boolean;
    dataRegistro: Date;
    
    erro: boolean;
    codigoErro: number | null;
    mensagemErro: string | null;
}

